modules/standard/Lists.chpl

```diff
@@ -0,0 +1,868 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+  This module contains the implementation of the List type.
+
+  A List is a lightweight container similar to an array that is suitable for
+  building up and iterating over a collection of elements in a structured
+  manner.
+
+  The highly parallel nature of Chapel means that great care should be taken
+  when performing operations that may invalidate references to List elements.
+  Inserts and removals into the middle of a List are example operations that
+  may invalidate references. Appending an element to the end of a List will
+  never invalidate references to elements contained in the List.
+
+  The following operations may invalidate references to elements contained in
+  a List:
+
+      - insert
+      - remove
+      - sort
+      - pop
+      - clear
+
+  Additionally, all references to List elements are invalidated when the List
+  is deinitialized.
+
+  List objects are not parallel safe by default, but can be made parallel safe
+  by setting the param formal `parSafe` to true in any List constructor. When
+  constructed from another List, a List object will inherit the parallel
+  safety mode of its originating List.
+
+  Inserts and removals into a List are O(n) worst case and should be performed
+  with care. Access into a List object is currently O(log n), but is expected
+  to reduce to O(1) in the future.
```

Comment:
Maybe say that Appends are amortized O(1) ?

Reply:
Isn't access O(1) right now?

Reply:
Yes.